requirements.txt

```diff
@@ -0,0 +1,42 @@
+#
+# These requirements were autogenerated by pipenv
+# To regenerate from the project's Pipfile, run:
+#
+#    pipenv lock --requirements
+#
+
+-i https://pypi.org/simple
+aniso8601==9.0.1; python_version >= '3.5'
```

Comment:
Same question as above, how are these versions generated (manually/auto)? For `requirements.txt` it seems a little rigid to pin exact dependency versions. @nishakm wdyt? In Tern, we use looser dependency versions for the requirements file in case a new version of a dependency comes out between releases, it doesn't interfere as much with development (see https://github.com/tern-tools/tern/blob/main/requirements.txt). For each release, however, we pin the deps as you've done so we have an exact record (i.e. https://github.com/tern-tools/tern/blob/main/docs/releases/v2_9_1-requirements.txt). This is not necessarily something that needs to change to merge this PR, just curious about both your thoughts on this when we think about the overrall workflow.

Reply:
So, it is handled by Pipenv.
Pipenv is a tool that helps manage the Python requirements versions conflicts for dependencies.
I used the pipenv to install the requirements, also the requirements-*.txt and the Pipfiles are managed by this tool.

Basically
```shell
pipenv install package
pipenv install -d package # only for development, ex: tox, black, etc
pipenv -l -r > requirements.txt # to update the requirements.txt
pipenv -l -r -d > requirements-dev.txt # to update the requirements-dev.txt
pipenv update # to update all packages
... 
```

We could also add to the CI to run checks.

Reply:
I forgot to mention.
The versions are not pinned once the entry point is the Pipfile. Only the black because it is beta versions, and the Pipenv protects you against using beta versions unless it is pinned. The requirements.txt is pinned automatically by Pipenv. It helps if you tag a release in the repository.

Reply:
It depends on whether we want to work on reducing the number of dependencies to the things we really need (similar to our `setup.cfg` work on the main project) now or later. It looks like most of the dependencies come from `flask-restx`. I lean towards merging now and trimming all the unused pieces later.